### PR TITLE
SQLStore: Fix SQLite error propagation if query retries are disabled

### PR DESCRIPTION
**Pull request by papagian**
**What is this feature?**

The following condition (introduced in #58559) used to fail if the query retries are disabled (equal to zero) (default behaviour):
https://github.com/grafana/grafana/blob/998b035d522af5e2a595d2086731d8aed09f83c2/pkg/services/sqlstore/session.go#L98
As a result due to retryer behaviour:
https://github.com/grafana/grafana/blob/998b035d522af5e2a595d2086731d8aed09f83c2/pkg/util/retryer/retryer.go#L43-L45
the `sqlite3.ErrLocked` and `sqlite3.ErrBusy` were not propagated

This fix changes the above condition to check for greater or equal instead

**Why do we need this feature?**

[Add a description of the problem the feature is trying to solve.]

**Who is this feature for?**

[Add information on what kind of user the feature is for.]

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

**Review by zserge — APPROVED** [review]

File: pkg/services/sqlstore/session_test.go [hunk 1]
```diff
@@ -10,6 +10,53 @@ import (
...
+			err := f(context.Background(), callback)
+			require.Error(t, err)
+			require.Equal(t, 1, i)
+		})
+
+		t.Run(fmt.Sprintf("%s should return the sqlite3.Error immediately", name), func(t *testing.T) {
+			i := 0
+			callback := func(sess *DBSession) error {
+				i++
+				return sqlite3.Error{Code: sqlite3.ErrBusy}
```
Comment (suntala) [inline pkg/services/sqlstore/session_test.go:38]: Should we also have a test case for the error being `sqlite3.ErrLocked`?

**Review by suntala — COMMENTED** [review]
Looks good to me overall! Just one small suggestion.

File: pkg/services/sqlstore/session_test.go [hunk 2]
```diff
@@ -44,6 +44,19 @@ func TestRetryingDisabled(t *testing.T) {
 			require.Equal(t, 1, i)
 		})
 
+		t.Run(fmt.Sprintf("%s should return the sqlite3.Error immediately", name), func(t *testing.T) {
```
Comment (suntala) [inline pkg/services/sqlstore/session_test.go:47]: Nit: Might be nice to change `sqlite3.Error` to `sqlite3.ErrBusy`/`sqlite3.ErrLocked` depending on the test case.

**Comment by papagian** [comment]
the bug was introduced in `9.4.0-beta1` (#58559) so no need to be backported to `9.3.x`

**Pull request merged**